orangecontrib/prototypes/widgets/owfeaturestatistics.py

```diff
@@ -0,0 +1,512 @@
+import locale
+from enum import auto, IntEnum
+from functools import partial
+from itertools import chain
+
+import numpy as np
+import scipy.stats as ss
+from AnyQt.QtCore import Qt, QSize, QRectF, QAbstractTableModel, QVariant, \
+    QModelIndex, pyqtSlot, QSortFilterProxyModel, QRegExp
+from AnyQt.QtGui import QPainter, QColor, QKeySequence
+from AnyQt.QtWidgets import QStyleOptionViewItem, QShortcut
+from AnyQt.QtWidgets import QStyledItemDelegate, QGraphicsScene, \
+    QTableView, QHeaderView, QStyle
+# pylint: disable=unused-import
+from typing import Any, Optional
+
+import Orange.statistics.util as ut
+from Orange.canvas.report import plural
+from Orange.data import Table, StringVariable, DiscreteVariable, \
+    ContinuousVariable, TimeVariable
+from Orange.widgets import widget, gui
+from Orange.widgets.settings import Setting, ContextSetting
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.utils.signals import Input
+from orangecontrib.prototypes.widgets.utils.histogram import Histogram
+
+
+class FeatureStatisticsTableModel(QAbstractTableModel):
+    DistributionRole = next(gui.OrangeUserRole)
+
+    CLASS_VAR, META, ATTRIBUTE = range(3)
+    COLOR_FOR_ROLE = {
+        CLASS_VAR: QColor(160, 160, 160),
+        META: QColor(220, 220, 200),
+        ATTRIBUTE: QColor(255, 255, 255),
+    }
+
+    class Columns(IntEnum):
+        ICON = auto()
+        NAME = auto()
+        DISTRIBUTION = auto()
+        FIRST_MOMENT = auto()
+        SECOND_MOMENT = auto()
+        MISSING = auto()
+
+        @property
+        def name(self):
+            return {self.ICON: '',
+                    self.NAME: 'Name',
+                    self.DISTRIBUTION: 'Distribution',
+                    self.FIRST_MOMENT: 'First moment',
+                    self.SECOND_MOMENT: 'Second moment',
```

Comment:
Use friendlier labels "Mean" and "Std".

Also compute "Min", "Max" and "Median" like `pandas.DataFrame.describe()` does.

Reply:
While I'd love to add these, quartiles would probably also make sense. However, these labels only make sense with ordinal variables and we need reasonable behaviour for nominal and date variables as well, where these values make little or no sense.

For discrete variables, I compute the mode and entropy for the first and second moment (these make some sort of sense?), and having the labels simply as "Mean" and "Std" would be confusing.

I have no good solution for this yet. Splitting them into separate columns would result in empty cells, which would be a complete waste space, that doesn't seem like a viable option.

Reply:
Isn't entropy as Second moment just as confusing?
Having mode and entropy for nominals is great, but the labels will be more intuitive (i.e. fewer StackExchange duplicates) if set to Mean and Std (or Center and Dispersion, or Mean/Mode and Std/Entropy). You can also tutor the user with `OWWidget.UserAdviceMessages` and header item tooltips. But I'll wager far more people understand the concepts mean, mode, std and (possibly) entropy than first and second moment. I'm normally strongly opposed to dumbing stuff down for idiots, but this really is making it complicated for no good reason. CC: @biolab/orange 

Min and Max _do_ make sense for datetime variables. For nominals, you could show the extreme labels from `Variable.values`. This is sensible when nominal `variable.ordered` (e.g. `few < more < many`). The widget is not too wide to add two more columns.

Re quartiles, they are not as essential as they can be (roughly) visually inferred from the histogram and the aforementioned reference points. 😄

Reply:
Yes, I agree, first and second moment is terrible labels. I really like Center and Dispersion.

Yes, min and max make sense for those cases, but in most cases the discrete variable is unordered.